### PR TITLE
Updated dev tools docker image tag to 0.4.6

### PR DESCRIPTION
**Pull request by paulpalamarchuk**
* Fixed lint tests
 * Updated Makefile
 * Fixed check_generate test function

### CODE REVIEW

**Comment by aaron-lane** [comment]
This was partially addressed by #292 which updated the image to 0.4.5.

File: modules/beta-private-cluster-update-variant/scripts/wait-for-cluster.sh [hunk 1]
```diff
@@ -15,6 +15,11 @@
 
 set -e
 
+# shellcheck disable=SC2034
+if [ -n "${GOOGLE_APPLICATION_CREDENTIALS}" ]; then
+    CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=${GOOGLE_APPLICATION_CREDENTIALS}
```
Comment (nick4fake) [inline modules/beta-private-cluster-update-variant/scripts/wait-for-cluster.sh]: Please add quotes

File: modules/beta-private-cluster-update-variant/main.tf [hunk 2]
```diff
@@ -34,6 +34,7 @@ resource "random_shuffle" "available_zones" {
 locals {
   // location
   location = var.regional ? var.region : var.zones[0]
+  region   = var.region == null ? join("-", slice(split("-", var.zones[0]), 0, 2)) : var.region
```
Comment (nick4fake) [inline modules/beta-private-cluster-update-variant/main.tf]: Are you sure it is safe?

File: modules/beta-private-cluster-update-variant/variables.tf [hunk 3]
```diff
@@ -267,6 +268,12 @@ variable "grant_registry_access" {
   default     = false
 }
 
+variable "registry_project_id" {
+  type        = string
+  description = "Project holding the Google Container Registry. If empty, we use the cluster project. If grant_registry_access is true, storage.objectViewer role is assigned on this project."
+  default     = ""
```
Comment (nick4fake) [inline modules/beta-private-cluster-update-variant/variables.tf]: Why do you have empty line as a default?

File: modules/beta-private-cluster/scripts/wait-for-cluster.sh [hunk 4]
```diff
@@ -15,6 +15,11 @@
 
 set -e
 
+# shellcheck disable=SC2034
+if [ -n "${GOOGLE_APPLICATION_CREDENTIALS}" ]; then
+    CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE=${GOOGLE_APPLICATION_CREDENTIALS}
```
Comment (nick4fake) [inline modules/beta-private-cluster/scripts/wait-for-cluster.sh]: same

File: modules/private-cluster-update-variant/variables.tf [hunk 5]
```diff
@@ -257,6 +258,12 @@ variable "grant_registry_access" {
   default     = false
 }
 
+variable "registry_project_id" {
+  type        = string
+  description = "Project holding the Google Container Registry. If empty, we use the cluster project. If grant_registry_access is true, storage.objectViewer role is assigned on this project."
+  default     = ""
```
Comment (nick4fake) [inline modules/private-cluster-update-variant/variables.tf]: same

File: modules/beta-private-cluster-update-variant/scripts/wait-for-cluster.sh (hunk 1 above)
Comment (paulpalamarchuk) [reply modules/beta-private-cluster-update-variant/scripts/wait-for-cluster.sh]: Added

**Comment by paulpalamarchuk** [comment]
PR rebased

**Pull request merged**